R/utilities-observed-data.R

```diff
@@ -85,6 +85,106 @@ readObservedDataFile <- function(fileName,
   return(observedData)
 }
 
+#' @title getSelectedData
+#' @description
+#' Get selected data 
+#' The function leverage `dplyr::filter` to select the data
+#' @param data A data.frame 
+#' @param dataSelection Character string or expression evaluated to select data
+#' The enum helper `DataSelectionKeys` provides keys for selected all or none of the data
+#' @return A data.frame of selected data
+#' @export
+#' @import dplyr
+#' @seealso DataSelectionKeys
+#' @examples
+#' data <- data.frame(
+#' x = seq(0,9),
+#' y = seq(10,19),
+#' mdv = c(1,1, rep(0, 8)),
+#' groups = rep(c("A", "B"), 5)
+#' )
+#' 
+#' # Select all the data
+#' getSelectedData(data, DataSelectionKeys$ALL)
+#' 
+#' # Select no data
+#' getSelectedData(data, DataSelectionKeys$NONE)
+#' 
+#' # Select data from group A
+#' getSelectedData(data, "groups %in% 'A'")
+#' 
+#' # Remove missing dependent variable (mdv)
+#' getSelectedData(data, "mdv == 0")
+#' 
+getSelectedData <- function(data, dataSelection) {
+  if(isEmpty(dataSelection)){
+    return(data[FALSE,])
+  }
+  if(isIncluded(dataSelection, DataSelectionKeys$ALL)){
+    return(data)
+  }
+  if(isIncluded(dataSelection, DataSelectionKeys$NONE)){
+    return(data[FALSE,])
+  }
+  if(isOfType(dataSelection, "expression")){
+    return(data %>% dplyr::filter(eval(dataSelection)))
+  }
+  return(data %>% dplyr::filter(eval(parse(text = dataSelection))))
+}
```

Comment:
Interesting, I did not think about that issue...
`isEmpty` checks the length of the object, I did not expect user input `dataSelection=""`.
In this case, I would provide the same behaviour as `dataSelection=NULL`.
Note that `dataSelection=NA` would select no data, because of the behaviour of the function `dplyr::filter`

```r
> length("")
[1] 1

> as.character(NULL)
character(0)

> isEmpty(as.character(NULL))
[1] TRUE

> is.null(as.character(NULL))
[1] FALSE
```

Reply:
ok, I created the issue for the isEmpty function in the ospsuite.utils